src/gui/qgssublayersdialog.cpp

```diff
@@ -42,50 +43,62 @@ class SubLayerItem : public QTreeWidgetItem
 };
 //! @endcond
 
-QgsSublayersDialog::QgsSublayersDialog( ProviderType providerType, const QString &name,
-                                        QWidget *parent, Qt::WindowFlags fl )
+QgsSublayersDialog::QgsSublayersDialog( ProviderType providerType,
+                                        const QString &name,
+                                        QWidget *parent,
+                                        Qt::WindowFlags fl,
+                                        const QString &providerSource )
   : QDialog( parent, fl )
   , mName( name )
 {
   setupUi( this );
   QgsGui::enableAutoGeometryRestore( this );
 
+  QString title;
   switch ( providerType )
   {
     case QgsSublayersDialog::Ogr :
-      setWindowTitle( tr( "Select Vector Layers to Add…" ) );
+      title = tr( "Select Vector Layers to Add…" );
       layersTable->setHeaderLabels( QStringList() << tr( "Layer ID" ) << tr( "Layer name" )
                                     << tr( "Number of features" ) << tr( "Geometry type" ) << tr( "Description" ) );
       mShowCount = true;
       mShowType = true;
       mShowDescription = true;
       break;
     case QgsSublayersDialog::Gdal:
-      setWindowTitle( tr( "Select Raster Layers to Add…" ) );
+      title = tr( "Select Raster Layers to Add…" );
       layersTable->setHeaderLabels( QStringList() << tr( "Layer ID" ) << tr( "Layer name" ) );
       break;
     case QgsSublayersDialog::Mdal:
-      setWindowTitle( tr( "Select Mesh Layers to Add…" ) );
+      title = tr( "Select Mesh Layers to Add…" );
       layersTable->setHeaderLabels( QStringList() << tr( "Layer ID" ) << tr( "Mesh name" ) );
       break;
     default:
-      setWindowTitle( tr( "Select Layers to Add…" ) );
+      title = tr( "Select Layers to Add…" );
       layersTable->setHeaderLabels( QStringList() << tr( "Layer ID" ) << tr( "Layer name" )
                                     << tr( "Type" ) );
       mShowType = true;
-      break;
   }
 
+  QString fileFullPath = providerType == QgsSublayersDialog::Vsifile
```

Comment:
Given that the argument is actually the dataSourceUri, I would call decodeUri here regardless of the providerType and if if contains "path" then use this.

Reply:
I agree and did it the way you said.